spec/controllers/challenge_assignments_controller_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe ChallengeAssignmentsController do
+  include LoginMacros
+  include RedirectExpectationHelper
+  let(:collection) { create(:collection) }
+
+  context "when not logged in" do
+    describe 'no_challenge' do
+      it 'error to login page' do
+        get :no_challenge, collection_id: collection.name
+        it_redirects_to_with_error(new_user_session_path, "Sorry, you don't have permission to access the page you were trying to reach. Please log in.")
+      end
+    end
+  end
+
+  context "when logged in" do
+    describe 'no_challenge' do
+      it 'show an error, redirect and return false' do
+        fake_login_known_user(@collection.owners.first.user)
```

Comment:
I think this is failing because this line and line 21 need to be `collection` (which is the one you've created with the `let`) instead of `@collection` (which doesn't exist)